### PR TITLE
[API] Hide archived shipping methods

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | Part of #11250
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php [hunk 1]
```diff
@@ -20,7 +20,6 @@
 use Sylius\Component\Addressing\Model\ZoneInterface;
 use Sylius\Component\Core\Model\ChannelInterface;
 use Sylius\Component\Core\Model\ShippingMethodInterface;
-use Symfony\Component\VarDumper\VarDumper;
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]: :trollface:

File: src/Sylius/Behat/Client/Request.php [hunk 2]
```diff
@@ -93,6 +93,11 @@ public static function delete(string $resource, string $id, string $token): Requ
     }
 
     public static function transition(string $resource, string $id, string $transition, string $token): RequestInterface
+    {
+        return self::customItemAction($resource, $id, $transition, $token);
+    }
+
+    public static function customItemAction(string $resource, string $id, string $transition, string $token): RequestInterface
```
Comment (Zales0123) [inline src/Sylius/Behat/Client/Request.php]: With such a change we assume that custom action is always a PATCH :/

File: src/Sylius/Bundle/ApiBundle/Controller/ArchiveShippingMethodAction.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Component\Core\Model\ShippingMethodInterface;
+
+final class ArchiveShippingMethodAction
+{
+    public function __invoke(ShippingMethodInterface $data): ShippingMethodInterface
+    {
+        $data->setArchivedAt(new \DateTime());
```
Comment (Zales0123) [inline src/Sylius/Bundle/ApiBundle/Controller/ArchiveShippingMethodAction.php]: How about using `Sylius\Bundle\ShippingBundle\Provider\Calendar::today()`?

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/HideArchivedShippingMethodExtension.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
```
Comment (Zales0123) [inline src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/HideArchivedShippingMethodExtension.php]: Missing license block

**Comment by Zales0123** [comment]
Thanks, Łukasz! :1st_place_medal:

**Pull request merged**